lib/Conversion/TorchToTosa/TorchToTosa.cpp

```diff
@@ -635,6 +635,56 @@ class ConvertAtenSqueezeAllDimsOp : public ConvertAtenSqueezeOp<AtenOpT> {
   }
 };
 
+// FIXME(AG): This will eventually go into a Tosa*Utils file
+// Convert an fp32 scalar into tosa fp32 tensor.
+static LogicalResult tosaF32TensorFromTorchFloat(ConversionPatternRewriter &rewriter,
+                                        Operation *op, Value torchScalarValue,
+                                        Value &tosaTensor) {
+  double scalarValue;
+  // Check if this is an unrealized conversion cast
+  auto uConvCast = dyn_cast<mlir::UnrealizedConversionCastOp>(torchScalarValue.getDefiningOp());
+  auto uConvCastInOps = uConvCast.inputs();
+  // Ensure that there is exactly one input
+  if (uConvCastInOps.front() != uConvCastInOps.back()) op->emitError("This cast must have exactly one operand");
+  auto torchScalarValue_ = uConvCast ? uConvCastInOps.front() : torchScalarValue;
+
+  if (!matchPattern(torchScalarValue_, m_TorchConstantFloat(&scalarValue)))
```

Comment:
I am not familiar with Tosa but it's a bit surprising that it can only handle the case the exp value is constant. Is it possible to also consider non constant value here?

Reply:
The tosa.pow operator expects two tensor parameters. The pow value could be broadcast. 

Torch offers a TorchTensor input and TorchScalar exponent here. Is there a simple construct to express the IntegerType MLIR::Value exponent into a scalar tensor, i.e. rank-0 TensorType with IntegerType element-type ?

Reply:
hmm, linalg can broadcast one scalar value to a tensor. However, this is done in the TorchToLinalg level.

Reply:
@cathyzhyi I will mark this as resolved.